### PR TITLE
Add tests and CI

### PR DESCRIPTION
**Pull request by ericpre**
- Add tests to check the creation of the shortcuts in windows and gnome,
- add travis and appveyor.

### CODE REVIEW

File: start_jupyter_cm/gnome.py [hunk 1]
```diff
@@ -54,8 +54,12 @@ def add_jupyter_here():
                 f.write(script % (PATH, terminal, PATH, terminal))
             st = os.stat(script_path)
             os.chmod(script_path, st.st_mode | stat.S_IEXEC)
-            call(['gio', 'set', '-t', 'string', '%s' % script_path,
-                  'metadata::custom-icon', 'file://%s' % logos[terminal]])
+            try:
```
Comment (francisco-dlp) [inline start_jupyter_cm/gnome.py]: Why is this needed? Currently the custom-icon registration doesn't work with the latest GNOME, but it doesn't raise an exception. Beside, skipping exceptions silently makes it harder to realise that there is an issue.

Comment (ericpre) [reply start_jupyter_cm/gnome.py]: This is some left over of a previous version to get the test to pass. This is removed now.

Comment (ericpre) [reply start_jupyter_cm/gnome.py]: Actually I was wrong, there is still a failure:
https://travis-ci.org/ericpre/start_jupyter_cm/builds/553717995

I will look at it later.

Comment (francisco-dlp) [reply start_jupyter_cm/gnome.py]: That's good because that's detecting the bug that I was talking about (the icon registration doesn't work with the latest version of GNOME). What's strange is that the tests pass with ``conda=True``

Comment (ericpre) [reply start_jupyter_cm/gnome.py]: > That's good because that's detecting the bug that I was talking about (the icon registration doesn't work with the latest version of GNOME).

The failure here is a different issue:
* when running from conda environment it can found `gio`, call it with the issue you mentioned, raise a warning and the test pass. In this case, gio is in the `folder` directory of the conda environment, which sounds a bit surprising to me.
* when running from the system python, the tests fail because it can't find `gio`... and it is indeed not in the PATH!

**Comment by ericpre** [comment]
https://travis-ci.org/ericpre/start_jupyter_cm/builds/554747277
https://ci.appveyor.com/project/ericpre/start-jupyter-cm/builds/25776512

**Comment by francisco-dlp** [comment]
It'll be good to merge this. @ericpre, could you solve the conflicts?

**Comment by francisco-dlp** [comment]
Where is travis?

**Comment by ericpre** [comment]
You need to activate it, I don't have the permission on travis, but it is broken at the minute! ;)

**Comment by ericpre** [comment]
It should be good now.
The travis build of my fork: https://travis-ci.org/github/ericpre/start_jupyter_cm/builds/702075442
